lib/azure/armrest/armrest_service.rb

```diff
@@ -252,13 +252,20 @@ def rest_head(options)
         end
 
         def raise_api_exception(err)
-          begin
-            response = JSON.parse(err.http_body)
-            code     = response['error']['code']
-            message  = response['error']['message']
-          rescue
-            code = err.try(:http_code) || err.try(:code)
-            message = err.try(:http_body) || err.try(:message)
+          content_type = err.response.try(:headers).try(:[], :content_type)
+          if content_type == "application/xml"
+            response = Hash.from_xml(err.http_body)
+            code     = response['Error']['Code']
+            message  = response['Error']['Message']
+          else
+            begin
+              response = JSON.parse(err.http_body)
+              code     = response['error']['code']
+              message  = response['error']['message']
+            rescue
+              code = err.try(:http_code) || err.try(:code)
+              message = err.try(:http_body) || err.try(:message)
+            end
```

Comment:
So I'm trying to minimize the behavioral change here but ideally I think this would look like:
```
case
err.response.headers[:content_type]
when "application/json"
  response = JSON.parse(err.http_body)
  code     = response['error']['code']
  message  = response['error']['message']
when "application/xml"
  response = Hash.from_xml(err.http_body)
  code     = response['Error']['Code']
  message  = response['Error']['Message']
else
  code = err.try(:http_code) || err.try(:code)
  message = err.try(:http_body) || err.try(:message)
end
```

This would require some changes to the specs though as there isn't normally a non-nil `err.response` in the RestClient exception that is built

Reply:
Okay I combined a bit of the switch on the content type with the rescue to catch any errors, seems like a safe-enough approach so I'm going to resolve this